### PR TITLE
'ReferenceError: AbortSignal is not defined' during jest execution

### PR DESCRIPTION
**Pull request by DariaGritsienko**
Fixes https://github.com/nodejs/undici/issues/1557

### CODE REVIEW

**Comment by mcollina** [comment]
@DariaGritsienko thanks for opening this! https://github.com/nodejs/undici/issues/2021 seems already fixed, are you sure you linked to the right PR?

**Comment by DariaGritsienko** [comment]
yes, thanks, fixed it - https://github.com/nodejs/undici/issues/1557
@mcollina

File: lib/fetch/request.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ const { URLSerializer } = require('./dataURL')
 const { kHeadersList } = require('../core/symbols')
 const assert = require('assert')
 const { setMaxListeners, getEventListeners, defaultMaxListeners } = require('events')
+const { AbortSignal } = require('node-abort-controller')
```
Comment (mcollina) [inline lib/fetch/request.js:33]: We should be using Node.js core `AbortSignal` if it is available

File: package.json [hunk 2]
```diff
@@ -69,6 +69,7 @@
   "devDependencies": {
     "@sinonjs/fake-timers": "^10.0.2",
     "@types/node": "^18.0.3",
+    "node-abort-controller": "^3.1.1",
```
Comment (mcollina) [inline package.json:72]: this must be a dependency.

**Review by mcollina — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
lgtm, let's see if this works!

**Comment by DariaGritsienko** [comment]
I studied this question a little more and found that jest adds in the jest-environment-node package global values, and they fixed it to auto in https://github.com/facebook/jest/pull/12642/files (v28.0.0)

`const contextGlobals = new Set(Object.getOwnPropertyNames(global));
    for (const nodeGlobalsKey of nodeGlobals) {
      if (!contextGlobals.has(nodeGlobalsKey)) {
        // @ts-expect-error
        global[nodeGlobalsKey] = globalThis[nodeGlobalsKey];
      }
    }`
    
I think undiсi will be better without this code... :) 

if people run into this problem they will need to install jest-environment-node package version >=28.0.0 and update jest >=28.0.0

I'll go try to update jest in my project, if anything else I'll be back

@mcollina

**Pull request closed without merging**